h5py/tests/hl/test_vds/test_virtual_source.py

```diff
@@ -0,0 +1,140 @@
+from ...common import ut
+import h5py as h5
+import numpy as np
+
+
+@ut.skipUnless(h5.version.hdf5_version_tuple >= (1, 9, 233),
+               'VDS requires HDF5 >= 1.9.233')
+class TestVirtualSource(ut.TestCase):
+    def test_full_slice(self):
+        dataset = h5.VirtualSource('test','test',(20,30,30))
+        sliced = dataset[:,:,:]
+        self.assertEqual(dataset.shape,sliced.shape)
+
+    # def test_full_slice_inverted(self):
+    #     dataset = h5.VirtualSource('test','test',(20,30,30))
+    #     sliced = dataset[:,:,::-1]
+    #     self.assertEqual(dataset.shape,sliced.shape)
+    #
+    # def test_subsampled_slice_inverted(self):
+    #     dataset = h5.VirtualSource('test','test',(20,30,30))
+    #     sliced = dataset[:,:,::-2]
+    #     self.assertEqual((20,30,15),sliced.shape)
```

Comment:
I commented out several tests in this file that were using either negative steps or `start > stop` reversed slices. These are rejected by the `SimpleSelection` machinery I switched to apply slices. I'm not sure if reversed slices are valid for VDS, but I wanted to prioritise getting something simple working, even if that means neglecting some corner cases.

Reply:
From the user point of view a VDS vs normal dataset should be identical.  I suspect we will want to handle the full set of 'fancy' slicing.  

That said, on access users should just see a normal dataset, this is only used to construct the VDS?  If that is the case, I agree that super fancy strides are important on a first pass at this.

Reply:
Yup, the `VirtualSource` class this was testing is only used when preparing a virtual dataset; access is just like a normal dataset. Though you can't access a dataset with a negative slice either (`ValueError: Step must be >= 1 (got -1)`).

I was debugging code added in the PR to translate Python slicing into an HDF5 selection, when I realised that h5py must already have that functionality to access slices of datasets. Sure enough, I found it in `h5py._hl.selections`. So I discarded the new code and used the selections module in its place. I suspect that if that module doesn't allow negative steps, it's because HDF5 selections don't support that.